### PR TITLE
Use headless Chrome in browser testing

### PR DESCRIPTION
**Pull request by kripken**
Instead of `xvfb-run`.

This is simpler, and I *hope* it fixes some of the big problems we've had with all tests failing and so forth. Some of that seems to have been tied to `xvfb` - maybe we were using it wrong, but aside from tests failing I've seen logs with tests running more than once, as if multiple browsers were running at the same time. Headless mode on the other hand seems much more stable, both locally and I did several runs on CI here.

One odd thing with headless mode is that the emrun test fails. Emrun loads another instance of chrome (separately from the test suite, that's what it does), and somehow in headless mode that instance stays around for a while longer. Instead of adding a `sleep` to try to work around that, I just made the second emrun invocation use another port (which also adds testing for the `--port` flag ;) )

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -166,9 +166,10 @@ test-chrome: &test-chrome
         command: |
           # --no-sandbox becasue we are running as root and chrome requires
```
Comment (dschuff) [inline .circleci/config.yml]: running as root? 😱

Comment (kripken) [reply .circleci/config.yml]: Hmm I think we ran as root in travis, but don't in circle. So maybe that's unnecessary, I can look into it later.

Comment (kripken) [reply .circleci/config.yml]: Turns out we do run in root. We don't request it, so I guess all circleCI jobs do? And that chrome bug is still present so we can't remove the no-sandbox flag...

**Comment by sbc100** [comment]
I seem to remember there were issues running GL tests with headless mode?   Are we currently running GL tests under chrome?

**Comment by kripken** [comment]
We haven't been running GL tests on chrome, but looks like we can now, I opened #9009

**Pull request merged**